docs/03_server/05_access-control/03_password_authentication.md

```diff
@@ -325,54 +325,116 @@ All requests below are capable of returning an error with a code of INTERNAL_ERR
-    MESSAGE_TYPE = EVENT_LOGIN_PREFS
-#### Response
-    MESSAGE_TYPE = EVENT_LOGIN_PREFS_ACK
-        DETAILS.PASSWORD_RESET_TYPE = ADMIN/EMAIL
+### Self-service password reset
+
+It is possible to initiate a self-service password message workflow assuming it is configured appropriately (see more [here](#selfServiceReset)).
+
+The message to request a self-service password reset looks like this:
+
+    MESSAGE_TYPE = EVENT_SELF_SERVICE_PASSWORD_RESET
+    DETAILS.USER_NAME = JohnWolf
+
+It is also possible to provide a RETURN_URL field as part of the details block if the URL used by the backend configuration is not suitable.
+
+Once this event has been triggered, the password reset link will be sent through the appropriate channels (e.g. e-mail) and the password reset action can be triggered.
+
+    MESSAGE_TYPE = EVENT_PASSWORD_RESET_ACTION
+    DETAILS.USER_NAME = JohnWolf
+    DETAILS.RESET_TOKEN = 1329048120a0sdf
+    DETAILS.NEW_PASSWORD = *******
+    DETAILS.INVALIDATE_ACTIVE_SESSIONS = true
+
+In this message sample we can identify the RESET_TOKEN field as the code provided to the user when the reset operation was requested, the NEW_PASSWORD field as the new password to be used for the user, and optionally, an INVALIDATE_ACTIVE_SESSIONS field to log out any currently logged in sessions.
+
+### Change password
+
+The change password message is available whether the user is currently logged in or not. This is important, as the first time a user logs in their newly created one-time password will be in a PASSWORD_EXPIRED state, which will force a "Change password" workflow before they can actually log in for the first time.
+
+The message structure for a change password message looks like this:
+
+    MESSAGE_TYPE = EVENT_CHANGE_USER_PASSWORD
+    DETAILS.USER_NAME = JohnWolf
+    DETAILS.OLD_PASSWORD = *******
+    DETAILS.NEW_PASSWORD = *******
+
+The naming convention for the fields is also self-explanatory.
+
+### Logout
+A logout request can be triggered before a user has logged in, in the case limits have been set for a maximum number of user sessions. This message workflow allows to terminate an existing active session so a new session can be created.
+
+A sample message would look like this:
+
+    MESSAGE_TYPE = EVENT_LOGOUT
+    DETAILS.USER_NAME = JohnWolf
+    DETAILS.SESSION_ID = *******
 
 ---
 
 ## Authentication
-Once you have a list of preferences, you can show the correct login dialog and let the user make a login attempt.  The password is provided in plain text, as it is expected that you will secure the connection using TLS.
+The password is provided in plain text, as it is expected that you will secure the connection using TLS.
 ### Login request
 
     MESSAGE_TYPE = EVENT_LOGIN_AUTH
     DETAILS.USER_NAME = JohnWolf
-    DETAILS.PASSWORD = FullMoon1
+    DETAILS.PASSWORD = ******
 
 ### Login response
-If successful:
-
-    MESSAGE_TYPE = EVENT_LOGIN_AUTH_ACK
-    DETAILS.SYSTEM.DATE = 2014-06-18 12:27:01
```

Comment:
2014... tells you where this example came from for sure and the reason why I had to really rework this page :)

Reply:
Hey, in my previous job, the system we were training people on started with (c)198-something

Reply:
but thanks for dragging my docs into the present day :)